tests/foreman/longrun/test_inc_updates.py

```diff
@@ -212,3 +212,83 @@ def test_positive_noapply_api(
         outval['action']
         == 'Incremental Update of 1 Content View Version(s) with 1 Package(s), and 1 Errata'
     )
+
+
+@pytest.mark.tier3
+def test_positive_incremental_update_time(module_target_sat, module_sca_manifest_org):
+    """Incremental update should not take a long time.
+
+    :id: a9cdcc58-2d10-42cf-8e24-f7bec3b79d6b
+
+    :steps:
+        1. Setup and sync rhel8 baseOS, rhst8, and rhsc8 repositories.
+        2. Create content view and add repos.
+        3. Publish a content view version with all content.
+        4. Using hammer, perform incremental update on that new version.
+        5. Publish the incremental version.
+
+    :expectedresults: Incremental update takes a short amount of time
+
+    :BZ: 2117760, 1829266
+
+    """
+    # rhel8 base os
+    rhel8_repo_id = module_target_sat.api_factory.enable_rhrepo_and_fetchid(
+        basearch=DEFAULT_ARCHITECTURE,
+        org_id=module_sca_manifest_org.id,
+        product=PRDS['rhel8'],
+        repo=REPOS['rhel8_bos']['name'],
+        reposet=REPOSET['rhel8_bos'],
+        releasever=REPOS['rhel8_bos']['releasever'],
+    )
+    rhel8_repo = module_target_sat.api.Repository(id=rhel8_repo_id).read()
+    assert rhel8_repo.sync(timeout=3000)['result'] == 'success'
+    # rh sat tools
+    rhst8_repo_id = module_target_sat.api_factory.enable_rhrepo_and_fetchid(
+        basearch=DEFAULT_ARCHITECTURE,
+        org_id=module_sca_manifest_org.id,
+        product=PRDS['rhel8'],
+        repo=REPOS['rhst8']['name'],
+        reposet=REPOSET['rhst8'],
+        releasever=None,
+    )
+    rhst8_repo = module_target_sat.api.Repository(id=rhst8_repo_id).read()
+    assert rhst8_repo.sync()['result'] == 'success'
+    # rh sat client
+    rhsc8_repo_id = module_target_sat.api_factory.enable_rhrepo_and_fetchid(
+        basearch=DEFAULT_ARCHITECTURE,
+        org_id=module_sca_manifest_org.id,
+        product=PRDS['rhel8'],
+        repo=REPOS['rhsclient8']['name'],
+        reposet=REPOSET['rhsclient8'],
+        releasever=None,
+    )
+    rhsc8_repo = module_target_sat.api.Repository(id=rhsc8_repo_id).read()
+    assert rhsc8_repo.sync()['result'] == 'success'
```

Comment:
May be, writing separate fixtures can be good way for this test module, this could reuse in any new additional test case.
btw there is an existing fixture for **Satellite client repo for RHEL7** (`sat_client_repo`) and can be use instead of `rhsclient8`

Reply:
We could also `.sync(synchronous=False)` to make the test faster (since the syncs would run in parallel) and do `wait_for_task` to ensure they succeeded.

Reply:
@vsedmik agree with `synch(synchronous=False)`, the lastest push includes the 3 repos in loop being added to CV and then sync started. Then we wait for repo sync tasks, poll succeeds, then perform incremental testing.

Reply:
@Griffin-Sullivan @vijaysawant for the implementation of a fixture, I didn't plan to use these repositories often, 
do we think a fixture would be needed for just this case ? 
For Griffin's recommendation, looping just once through the repo_ids, latest push has this refactored. 
Let me know how the changes look!

Reply:
@Griffin-Sullivan Are you fine with the changes? Can we resolve this thread?